src/helpers/misc.ts

```diff
@@ -248,3 +248,33 @@ export function isDefined<T>(argument: T | undefined): argument is T {
 }
 
 export const DEBUG: { [key: string]: any } = {};
+
+/**
+ * Compares two objects.
+ */
+export function deepEquals(o1: any, o2: any): boolean {
+  if (o1 === o2) return true;
+  if ((o1 && !o2) || (o2 && !o1)) return false;
+  if (typeof o1 !== typeof o2) return false;
+  if (typeof o1 !== "object") return o1 === o2;
+
+  // Objects can have different keys if the values are undefined
+  const keys = new Set<string>();
+  Object.keys(o1).forEach((key) => keys.add(key));
+  Object.keys(o2).forEach((key) => keys.add(key));
+
+  for (let key of keys) {
+    if (typeof o1[key] !== typeof o1[key]) return false;
+    if (typeof o1[key] === "object") {
+      if (!deepEquals(o1[key], o2[key])) return false;
+    } else {
+      if (o1[key] !== o2[key]) return false;
+    }
+  }
+
+  return true;
+}
+
+export function areConditionalFormatEquivalents(cf1: ConditionalFormat, cf2: ConditionalFormat) {
```

Comment:
I'd keep it inside the CF file if it's used only there (and it's quite specific to CF)